File: Day123_2025/Q123_Leetcode.cpp
class Solution {
    public:
        int minDominoRotations(vector<int>& tops, vector<int>& bottoms) {
            // Q. Minimum Domino Rotations For Equal Row
            vector<int> countA(7, 0), countB(7, 0), countSame(7, 0);
            for(int i = 0; i < tops.size(); i++){
                countA[tops[i]]++;
                countB[bottoms[i]]++;
                if(tops[i] == bottoms[i]) countSame[tops[i]]++;
            }
            int n = tops.size(), ans = n;
            for(int i = 1; i <= 6; i++){
                if(countA[i] + countB[i] - countSame[i] == n){
                    ans = min(ans, n - max(countA[i], countB[i]));
                }
            }
            return ans == n ? -1 : ans;
        }
    };

File: Day123_2025/Q123_Leetcode.py
from typing import List


class Solution:
    def minDominoRotations(self, tops: List[int], bottoms: List[int]) -> int:
        countA = [0] * 7
        countB = [0] * 7
        countSame = [0] * 7
        n = len(tops)

        for i in range(n):
            countA[tops[i]] += 1
            countB[bottoms[i]] += 1
            if tops[i] == bottoms[i]:
                countSame[tops[i]] += 1

        ans = n
        for i in range(1, 7):
            if countA[i] + countB[i] - countSame[i] == n:
                ans = min(ans, n - max(countA[i], countB[i]))

        return -1 if ans == n else ans
